src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs

```diff
@@ -687,7 +742,7 @@ private void SendDiagnostics(IList<DiagnosticsMessage> diagnostics)
                 {
                     ContextId = Id,
                     MessageType = "AllDiagnostics",
-                    Payload = JToken.FromObject(diagnostics)
+                    Payload = JToken.FromObject(diagnostics.Select(d => d.ConvertToJson(_protocolVersion)).ToList())
```

Comment:
Store this in a variable instead of doing it inline. Also pass ProtocolVersion instead of _protocolVersion